src/codeu/chat/server/Server.java

```diff
@@ -176,6 +177,19 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
       }
     });
 
+       // New Interest - A client wants to add a new interest to the back end.
+    this.commands.put(NetworkCode.NEW_INTERESTS_REQUEST,  new Command() {
+      @Override
+      public void onMessage(InputStream in, OutputStream out) throws IOException {
+
+        final Collection<Uuid> ids = Serializers.collection(Uuid.SERIALIZER).read(in);
+        final Interests interests = controller.newInterests(ids);
```

Comment:
There's also no way for you to know, but I named this `newInterest(Uuid userid, Uuid interest)` not newInterests(Set<Uuids> ids) because you're adding a single interest, not an Interests object necessarily.